homeassistant/components/tplink_eap/__init__.py

```diff
@@ -0,0 +1,57 @@
+"""The TP-Link EAP integration."""
+import asyncio
+
+from pytleap import Eap
+
+from homeassistant.components.device_tracker.const import (
+    DOMAIN as DEVICE_TRACKER_DOMAIN,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN
+
+PLATFORMS = [DEVICE_TRACKER_DOMAIN]
+
+
+async def async_setup(hass: HomeAssistant, config):
+    """Set up the TP-Link EAP component.
+
+    Configuration through YAML is not supported at this time.
+    """
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up TP-Link EAP from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
Is it expected that we initialize the data for the domain as currently done? It was not present in the scaffolded files and without it, `hass.data[DOMAIN]` raises a `KeyError` when trying to store the API object.

Reply:
This is fine 👍